### DISCUSSION TITLE
completely lost, need dumbed down instructions

### DISCUSSION THREAD

**Discussion by sinkust** [UltimaHoarder/UltimaScraper#1589] [General]
My onlyfans subscriptions are going to run out before I figure this out on my own.. 

The instructions are probably pretty clear to someone who knows a thing or two about that kind of stuff, but it might as well be in a different language for me

**Comment by vacuran** [comment]
How far have you gotten? What error are you receiving? The directions can be a little unclear at times, but your question of just explain everything comes across as though you've not tried anything.

**Reply by jobob3r** [reply]
What are the instructions on windows? I get "install python" but I need more direction than that, and the readme is just a copy of the main github page, which doesn't have the info I need

Step 1 is install python and then...what? run from powershell? command prompt? double click on the start_ofd.py file? open up _python's_ terminal and run it through that? do I need to run some command to change the directory within python and then change it to that? the instructions are not clear.

**Comment by LunarPenguin** [comment]
Have a look at #1542

**Reply by sinkust** [reply]
`py -3 -m pip install -U -r requirements.txt` gives me invalid sytax under pip 

Fixed that by opening a new powershell window..


Now step 7 returns `No module named 'sq1a1chemy'`

**Reply by vacuran** [reply]
Try running `py -3 -m pip install -U -r requirements.txt` again. If everything installed correctly you should get a giant list of things saying requirement already satisfied and if anything isn't installed it will try and install it. 

If it is still giving an invalid syntax error then confirm python is installed by running `python --version` and make sure that python installed correctly.

I can't imagine this to make a difference, but you can also make sure that PIP is upgraded by doing `pip install --upgrade pip`. If it does actually upgrade you'll want to close your PowerShell window and reopen before trying from step 5 again.

**Reply by LunarPenguin** [reply]
you can run `py -3 -m pip freeze` this returns a list of user installed modules, make sure `sqlalchemy` is in that list. if not run `py -3 -m pip install -U SQLAlchemy`

**Reply by sinkust** [reply]
that's all done and I've filled out the auth.json file, and im getting back `Site: 0 = onlyfans `

**Reply by americanseeder1865** [reply]
alright, press the 0 key, then the 1 key. I am going to guess you're likely to run into some auth issues, but i can help a bit.

**Reply by sinkust** [reply]
after following all the prompts It starts to scrap then I get 

`TypeError: object list can't be used in 'await' expression`

**Reply by americanseeder1865** [reply]
I need more context here, can I get the full error? make sure to blot out personal information. I do not care to know who you are.

**Reply by americanseeder1865** [reply]
I am also guessing this is a known bug too, can you also post what your requirements.txt file looks like

**Reply by sinkust** [reply]
```
Type: Profile
Type: Stories
No Stories Found.
Type: Posts
Scrape Attempt: 1/100
Type: Archived Posts
Scrape Attempt: 1/100
Processing Scraped Posts
  0%|                                                                                            | 0/1 [00:00<?, ?it/s]Traceback (most recent call last):
  File "E:\*****\OnlyFans-7.4.1\start_ofd.py", line 60, in <module>
    loop.run_until_complete(main())
  File "C:\Users\*****\AppData\Local\Programs\Python\Python39\lib\asyncio\base_events.py", line 642, in run_until_complete
    return future.result()
  File "E:\*****\OnlyFans-7.4.1\start_ofd.py", line 44, in main
    api = await main_datascraper.start_datascraper(
  File "E:\Jasons Folder\OnlyFans-7.4.1\datascraper\main_datascraper.py", line 93, in start_datascraper
    await main_helper.process_names(
  File "E:\*****\OnlyFans-7.4.1\helpers\main_helper.py", line 939, in process_names
    result = await module.start_datascraper(authed, username, site_name)
  File "E:\*****\OnlyFans-7.4.1\modules\onlyfans.py", line 160, in start_datascraper
    await prepare_scraper(authed, site_name, item)
  File "E:\*****\OnlyFans-7.4.1\modules\onlyfans.py", line 823, in prepare_scraper
    unrefined_set = await tqdm.gather(tasks, **settings)
  File "C:\Users\*****\AppData\Local\Programs\Python\Python39\lib\site-packages\tqdm\asyncio.py", line 79, in gather
    res = [await f for f in cls.as_completed(ifs, loop=loop, timeout=timeout,
  File "C:\Users\*****\AppData\Local\Programs\Python\Python39\lib\site-packages\tqdm\asyncio.py", line 79, in <listcomp>
    res = [await f for f in cls.as_completed(ifs, loop=loop, timeout=timeout,
  File "C:\Users\*****\AppData\Local\Programs\Python\Python39\lib\asyncio\tasks.py", line 614, in _wait_for_one
    return f.result()  # May raise f.exception().
  File "C:\Users\*****\AppData\Local\Programs\Python\Python39\lib\site-packages\tqdm\asyncio.py", line 76, in wrap_awaitable
    return i, await f
TypeError: object list can't be used in 'await' expression
```

**Reply by sinkust** [reply]
requirements.txt has 

```
aiohttp
aiohttp_socks
alembic
beautifulsoup4
jsonpickle
lxml
mergedeep
psutil
python-dateutil
python-socks[asyncio]
requests
selenium
selenium-wire==2.1.2
sqlalchemy==1.4.20
tqdm
ujson
urllib3
user_agent
win32-setctime
```

**Reply by LunarPenguin** [reply]
Which version of `tqdm` is installed? if it is `4.62.0` try downgrading to `4.59.0` and check if it solves your problem.

**Reply by sinkust** [reply]
how do I do that?

**Reply by LunarPenguin** [reply]
- to show which version you have installed run `py -3 -m pip show tqdm`
- to remove `tqdm` run `py -3 -m pip uninstall tqdm`
- to install `tqdm` version `4.59.0` run `py -3 -m pip install -U tqdm==4.59.0`

**Reply by sinkust** [reply]
it's working now, thank you

**Reply by LunarPenguin** [reply]
> it's working now, thank you

That's good to hear

**Reply by androidfan84** [reply]
I just wanted to thank everyone on this thread. I had many of the same issues here and this was the only thread that actually helped me step through all the issues I was having. Also on a side note.. Python 3.10 WILL NOT work for this as of writing this reply. It's missing many libraries and functions that caused me a nightmare attempting to troubleshoot just to get this script working. I ended up going back to the last 3.9 release and using this thread to fix tqdm got the rest working.

Lastly, one thing that was not abundantly clear to me was pressing 0 on the initial menu when running the script.. ignorantly I expected it to be automated until the initial prompt.

**Reply by LunarPenguin** [reply]
That's nice to hear. 

Regarding python 3.10: Development on this version of the repo has basically stalled because the maintainer currently has no access to the account this repo is on and has thus has created an alt account @DIGITALCRIMINALS which forks this repo and python 3.10 is supported on that repo.
